bls/src/main/java/tech/pegasys/teku/bls/impl/blst/BlstSignature.java

```diff
@@ -57,21 +55,16 @@ static BlstSignature fromSignature(Signature signature) {
   }
 
   public static BlstSignature aggregate(List<BlstSignature> signatures) {
-
-    Optional<BlstSignature> invalidSignature =
-        signatures.stream().filter(s -> !s.isValid).findFirst();
-    if (invalidSignature.isPresent()) {
-      throw new IllegalArgumentException(
-          "Can't aggregate invalid signature: " + invalidSignature.get());
-    }
-
-    // We've done the group check, so we can add() rather than aggregate() here
-    P2 sum = new P2();
-    for (BlstSignature finiteSignature : signatures) {
-      sum.add(finiteSignature.ec2Point);
+    try {
+      P2 sum = new P2();
+      for (BlstSignature finiteSignature : signatures) {
+        sum.aggregate(finiteSignature.ec2Point);
+      }
+      return new BlstSignature(sum.to_affine());
+    } catch (RuntimeException e) {
+      // Blst performs a G2 group membership test on each signature. We end up here if it fails.
+      throw new RuntimeException("Failed to aggregate signatures: " + e.getMessage());
```

Comment:
I'm really not a fan of throwing a generic `RuntimeException` here, it's just too generic and you wind up catching `RuntimeException` all over the place then, potentially sweeping up exceptions you didn't intend to catch. I'd probably rename `DeserializeException` to `BlsException` so we can throw it here.  And worth keeping the full details of the original exception too with `new BlsException("Failed to aggregate signatures", e)`

Reply:
Good plan - done. I've reworked exceptions generally across the BLS classes. It's not perfect, but simpler than it was.